### PR TITLE
Added instructions for local debugging via Docker Instance and Remote Debugger

### PR DESCRIPTION
**Pull request by ravidesai47**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -29,4 +29,21 @@ $CODE/statesman-server/config/local.yml
...
 
-
+To debug the statesman service running via Docker in IntelliJ, follow below mentioned steps:
+```shell
+# Uncomment below mentioned line from $CODE/statesman-server/Dockerfile
+EXPOSE 8084
+
+# Uncomment below mentioned lines from $CODE/statesman-server/docker-compose.yml
+- "8084:8084"
+- JAVA_OPTS: -agentlib:jdwp=transport=dt_socket,server=y,suspend=n,address=8084
```
Comment (deadmangareader) [inline README.md:39]: I suggest we should rename JAVA_OPTS to DEBUG_JAVA_OPTS, to call out its usage only for debug

File: statesman-server/docker-compose.yml [hunk 2]
```diff
@@ -20,6 +20,7 @@ services:
     ports:
     - "8080:8080"
     - "8081:8081"
+    # - "8084:8084"
```
Comment (deadmangareader) [inline statesman-server/docker-compose.yml:23]: Are we using compose file for production image? If not, we don't have to comment out stuff.

File: README.md (hunk 1 above)
Comment (ravidesai47) [reply README.md:39]: Sounds good. I will update the same.

File: statesman-server/docker-compose.yml (hunk 2 above)
Comment (ravidesai47) [reply statesman-server/docker-compose.yml:23]: According to me let's keep this commented and developer uncomment it as and when required.

File: README.md (hunk 1 above)
Comment (ravidesai47) [reply README.md]: Updated

File: statesman-server/docker-compose.yml (hunk 2 above)
Comment (santanusinha) [reply statesman-server/docker-compose.yml]: Why? Having the port exposed hurts nothing. OTOH this way, it adds a setp to the process

File: README.md [hunk 3]
```diff
@@ -22,11 +22,28 @@ docker-compose build
...
 ```
 
-
+To debug the statesman service running via Docker in IntelliJ, follow below mentioned steps:
+```shell
+# Uncomment below mentioned line from $CODE/statesman-server/Dockerfile
+EXPOSE 8090
+
+# Uncomment below mentioned lines from $CODE/statesman-server/docker-compose.yml
+- "8090:8090"
```
Comment (santanusinha) [inline README.md:38]: Not needed. Uncomment this

File: statesman-server/Dockerfile [hunk 4]
```diff
@@ -4,8 +4,7 @@ RUN apt-get clean && apt-get update && apt-get install -y --no-install-recommend
 
 EXPOSE 8080
 EXPOSE 8081
-EXPOSE 8082
-EXPOSE 8083
+# EXPOSE 8090
```
Comment (santanusinha) [inline statesman-server/Dockerfile:7]: Uncomment this

File: statesman-server/docker-compose.yml [hunk 5]
```diff
@@ -29,6 +30,7 @@ services:
       DB_PASSWORD: password
       FOXTROT_ENDPOINT: https://localhost:17000
       FOXTROT_TOKEN: INV_TOKEN_
+      # JAVA_DEBUG_OPTS: -agentlib:jdwp=transport=dt_socket,server=y,suspend=n,address=8090
```
Comment (santanusinha) [inline statesman-server/docker-compose.yml:33]: The compose is only used for local dev. This can be enabled always. Uncomment this

**Review by santanusinha — CHANGES_REQUESTED** [review]

File: statesman-server/docker-compose.yml (hunk 2 above)
Comment (ravidesai47) [reply statesman-server/docker-compose.yml]: Uncommented relevant code

File: README.md (hunk 3 above)
Comment (ravidesai47) [reply README.md]: Done

File: statesman-server/Dockerfile (hunk 4 above)
Comment (ravidesai47) [reply statesman-server/Dockerfile]: Done

File: statesman-server/docker-compose.yml (hunk 5 above)
Comment (ravidesai47) [reply statesman-server/docker-compose.yml]: Done

**Review by santanusinha — APPROVED** [review]

**Pull request merged**